cob_obstacle_distance/src/cob_obstacle_distance.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_obstacle_distance
+ *
+ * \author
+ *   Author: Marco Bezzon, email: Marco.Bezzon@ipa.fraunhofer.de
+ *
+ * \date Date of creation: May, 2015
+ *
+ * \brief
+ *   Main to initialize a node and start publishing markers.
+ *
+ ****************************************************************/
+#include <ctime>
+#include <vector>
+#include <ros/ros.h>
+#include <fcl/shape/geometric_shapes.h>
+#include <fstream>
+
+#include "cob_obstacle_distance/distance_manager.hpp"
+#include "cob_obstacle_distance/marker_shapes.hpp"
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "cob_obstacle_distance");
+    ros::NodeHandle nh("cob_obstacle_distance");
+    ros::Rate r(1.0);
+
+    DistanceManager sm(nh);
+    if (0 != sm.init())
+    {
+        ROS_ERROR("Failed to initialize DistanceManager.");
+        return -4;
+    }
+
+    ros::ServiceServer registration_srv = nh.advertiseService(sm.getRoboNamespace() + "/registerPointOfInterest" , &DistanceManager::registerPointOfInterest, &sm);
```

Comment:
Do you only use robo_namespace to advertise the service?
Then we definitely should use the ```ns="arm_right"``` tag in the launch file and a a relative name for advertisment here
```nh.advertiseService("registerPointOfInterest" , &DistanceManager::registerPointOfInterest, &sm);```

have a look at using ```ros::NodeHandle()``` and ros::NodeHandle("namespace")``` for parameter parsing and subscribing/advertising in [cob_twist_controller](https://github.com/ipa320/cob_control/blob/indigo_dev/cob_twist_controller/src/cob_twist_controller.cpp#L43-L144)